examples/tf/te_ppo_metaworld_ml1_push.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python3
+"""This is an example to train Task Embedding PPO with PointEnv."""
+# pylint: disable=no-value-for-parameter
+import akro
+import click
+from metaworld.benchmarks import ML1
+import numpy as np
+import tensorflow as tf
+
+from garage import InOutSpec, wrap_experiment
+from garage.envs import GarageEnv, normalize
+from garage.envs.multi_env_wrapper import MultiEnvWrapper
+from garage.experiment.deterministic import set_seed
+from garage.np.baselines import LinearMultiFeatureBaseline
+from garage.sampler import LocalSampler
+from garage.tf.algos import TEPPO
+from garage.tf.algos.te import TaskEmbeddingWorker
+from garage.tf.embeddings import GaussianMLPEncoder
+from garage.tf.experiment import LocalTFRunner
+from garage.tf.policies import GaussianMLPTaskEmbeddingPolicy
+
+
+@click.command()
+@click.option('--seed', default=1)
+@click.option('--n_epochs', default=600)
+@click.option('--batch_size_per_task', default=1024)
+@wrap_experiment
+def te_ppo_ml1_push(ctxt, seed, n_epochs, batch_size_per_task):
+    """Train Task Embedding PPO with PointEnv.
+
+    Args:
+        ctxt (garage.experiment.ExperimentContext): The experiment
+            configuration used by LocalRunner to create the snapshotter.
+        seed (int): Used to seed the random number generator to produce
+            determinism.
+        n_epochs (int): Total number of epochs for training.
+        batch_size_per_task (int): Batch size of samples for each task.
+
+    """
+    set_seed(seed)
+    envs = [GarageEnv(normalize(ML1.get_train_tasks('push-v1')))]
+    env = MultiEnvWrapper(envs, mode='del-onehot')
+
+    latent_length = 4
+    inference_window = 6
+    batch_size = batch_size_per_task
+    policy_ent_coeff = 2e-2
+    encoder_ent_coeff = 2.e-4
+    inference_ce_coeff = 5e-2
+    max_path_length = 100
+    embedding_init_std = 0.01
+    embedding_max_std = 0.02
+    embedding_min_std = 1e-6
+    policy_init_std = 1.0
+    policy_max_std = None
+    policy_min_std = None
+
+    with LocalTFRunner(snapshot_config=ctxt) as runner:
+
+        latent_lb = np.zeros(latent_length, )
+        latent_ub = np.ones(latent_length, )
+        latent_space = akro.Box(latent_lb, latent_ub)
+
+        # trajectory space is (TRAJ_ENC_WINDOW, act_obs) where act_obs is a
+        # stacked vector of flattened actions and observations
+        obs_lb, obs_ub = env.observation_space.bounds
+        obs_lb_flat = env.observation_space.flatten(obs_lb)
```

Comment:
can you add a helper classmethod to TEPPO to make this less painful? See PEARL and RL2 for examples.